File: dsa/6.hash_table/1_Two_Sum.py
import typing as t

class Solution:
    def twoSum(self, nums: t.List[int], target: int) -> t.List[int]:
        unique_dict: t.Dict[int, int] = {}
        for ind in range(0, len(nums), 1):
            if nums[ind] in unique_dict.keys():
                return [ind, unique_dict.get(nums[ind])]
            else:
                unique_dict[target - nums[ind]] = ind
        
        return [-1, -1]

File: dsa/6.hash_table/1_Two_Sum.java
package hash_table;
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> valMap = new HashMap<Integer, Integer>();
        int[] result = new int[2];
        result[0] = -1;
        result[1] = -1;
        for (int i = 0; i < nums.length; i++){
            if (valMap.containsKey(target-nums[i]) == true){
                result[0] = valMap.get(target-nums[i]);
                result[1] = i;
                return result;
            }
            valMap.put(nums[i], i);
        }
        return result;
    }
}
